src/HappyEyeBallsConnectionBuilder.php

```diff
@@ -0,0 +1,304 @@
+<?php
+
+namespace React\Socket;
+
+use React\Dns\Model\Message;
+use React\Dns\Resolver\Resolver;
+use React\EventLoop\LoopInterface;
+use React\EventLoop\TimerInterface;
+use React\Promise;
+use React\Promise\CancellablePromiseInterface;
+use InvalidArgumentException;
+use RuntimeException;
+
+/**
+ * @internal
+ */
+final class HappyEyeBallsConnectionBuilder
+{
+    public $loop;
+    public $connector;
+    public $resolver;
+    public $uri;
+    public $host;
+    public $resolved = array();
+    public $resolverPromises = array();
+    public $connectionPromises = array();
+    public $connectQueue = array();
+    public $timer;
+    public $parts;
+    public $ipsCount = 0;
+    public $failureCount = 0;
+    public $resolve;
+    public $reject;
+
+    public static function create(LoopInterface $loop, ConnectorInterface $connector, Resolver $resolver, $uri)
+    {
+        $connectionBuilder = new self($loop, $connector, $resolver, $uri);
+
+        return $connectionBuilder->connect();
+    }
```

Comment:
It looks like this class is only used internally anyway, so why not use the constructor directly?

Reply:
Done :+1: